support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -161,32 +184,85 @@ func (r *PodReconciler) updateorCreatePodEntry(ctx context.Context, pod *corev1.
 }
 
 // podSpiffeID returns the desired spiffe ID for the pod, or nil if it should be ignored
-func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) string {
+func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) (string, error) {
 	if r.c.PodLabel != "" {
 		// the controller has been configured with a pod label. if the pod
 		// has that label, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if labelValue, ok := pod.Labels[r.c.PodLabel]; ok {
-			return makeID(r.c.TrustDomain, "%s", labelValue)
+			return makeID(r.c.TrustDomain, "%s", labelValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	if r.c.PodAnnotation != "" {
 		// the controller has been configured with a pod annotation. if the pod
 		// has that annotation, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if annotationValue, ok := pod.Annotations[r.c.PodAnnotation]; ok {
-			return makeID(r.c.TrustDomain, "%s", annotationValue)
+			return makeID(r.c.TrustDomain, "%s", annotationValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	// the controller has not been configured with a pod label or a pod annotation.
+	if r.c.IdentityTemplate != "" {
+		// create an entry using provided identity template.
+		svid, err := r.getIdentityTemplate(pod)
+		if err != nil {
+			return "", err
+		}
+		return makeID(r.c.TrustDomain, svid), nil
+	}
```

Comment:
I think we need to gate this similar to the PodAnnotation/PodLabel. ie if the pod has label `spiffe.io/spiffeid` then we generate the the entry using the `IdentityTemplate`. Without the gate it'll generate an entry for every pod in the system.

Reply:
> I think we need to gate this similar to the PodAnnotation/PodLabel. ie if the pod has label `spiffe.io/spiffeid` then we generate the the entry using the `IdentityTemplate`. Without the gate it'll generate an entry for every pod in the system.

This gets back to my original point about Scope.  I may want this for every Pod in the cluster (I guarantee there will be folks who want that).  I may want this for every Pod in my Namespace, or Namespace+ServiceAccount.  I may want to pick and choose which Pods I want to use this template for.

What I would really really not want is to have to annotate *every* Pod *everytime* if my desired outcome is 'all in cluster/namespace/serviceaccount'...

Reply:
i see how that can be a drag. how about a configurable `IdentityTemplateLabel`? If its empty generate the entry for every pod in the cluster. If its non-empty restrict to only Pods with the label.

Reply:
@faisal-memon Again... that takes things back to global... so I can either have identities issued to *every* Pod in my cluster (sometimes desireable, but not always) or I can require every Pod spec for every app to be altered (sometimes but not always doable).  The root issue isn't 'annotations' vs 'labels'... its 'change every Pod' vs selection of Pods to which things should be applied.

K8s (and all of Cloud Native) tends heavily away from 'enumeration' (which is what change per Pod is) and towards selection.

Reply:
What do you suggest?

Reply:
So per agreement during our contributors call, we will create an `IdentityTemplateLabel` parameter in the registrar configuration(CRD mode only). All the pods with this label set to `true` would get the identity defined in the custom template.  If the value of this parameter is empty, the template format would apply to all the pods. 
As a next step, we would focus on scoping it on the level of the namespace, service-accounts, or both.

Reply:
done